modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -2840,7 +2840,7 @@ void GDScriptAnalyzer::reduce_call(GDScriptParser::CallNode *p_call, bool p_is_a
 			parser->push_warning(p_call, GDScriptWarning::RETURN_VALUE_DISCARDED, p_call->function_name);
 		}
 
-		if (is_static && !base_type.is_meta_type && !(is_self && parser->current_function != nullptr && parser->current_function->is_static)) {
+		if (is_static && !base_type.is_meta_type && !(is_self && parser->current_function != nullptr && parser->current_function->is_static) && callee_type != GDScriptParser::Node::IDENTIFIER) {
```

Comment:
This doesn't permit the use of the`self` identifier.

Reply:
Like was mentioned [here](https://github.com/godotengine/godot/issues/69282#issuecomment-1329962963), in other languages (in the comment it is C#) it is allowed to call static functions just by identifier from inside a class, but not allowed to do so by calling `instance.stat()` or `this.stat()`. If the intention is to be like other languages then makes sense to keep the warning for `self.stat()`.